File: frontend/lib/operations.ts
import { api } from "./api";

export type PruneReport = {
  started_at: string;
  finished_at: string;
  audit_logs_deleted: number;
  collector_runs_deleted: number;
  rollups_optimized: string[];
  disk_percent: number;
  disk_free_bytes: number;
  watchdog_fired: boolean;
  actions_taken: string[];
  errors: string[];
};

export type ReclaimEstimate = {
  audit_logs_rows_to_delete: number;
  collector_runs_rows_to_delete: number;
  failed_inserts_rows: number;
  docker_hint: string;
};

export const operationsApi = {
  lastPrune:        () => api<PruneReport | null>("/operations/last-prune"),
  runPrune:         () => api<PruneReport>("/operations/prune", { method: "POST" }),
  reclaimEstimate:  () => api<ReclaimEstimate>("/operations/reclaim-estimate"),
};

File: backend/app/schemas/operations.py
from datetime import datetime
from pydantic import BaseModel


class PruneReportOut(BaseModel):
    started_at: datetime
    finished_at: datetime
    audit_logs_deleted: int
    collector_runs_deleted: int
    rollups_optimized: list[str]
    disk_percent: float
    disk_free_bytes: int
    watchdog_fired: bool
    actions_taken: list[str]
    errors: list[str]


class ReclaimEstimate(BaseModel):
    """Best-effort estimate of bytes that scripts/reclaim.sh could free."""
    audit_logs_rows_to_delete: int
    collector_runs_rows_to_delete: int
    failed_inserts_rows: int
    docker_hint: str
